src/main/java/com/ably/kafka/connect/ChannelSinkTask.java

```diff
@@ -48,6 +57,19 @@ public void put(Collection<SinkRecord> records) {
         }
 
         for (final SinkRecord record : records) {
+            publishSingleRecord(record);
+        }
+    }
+
+    private void publishSingleRecord(SinkRecord record) {
+        SinkRecord suspendRecord = null;
+        if (suspended.get()){
+            suspendQueue.enqueue(record);
+        } else if ((suspendRecord = suspendQueue.dequeue()) != null) {
+            ablyClient.publishFrom(suspendRecord);
+            //re-call for the same record so that (if available) other suspended records are processed first
+            publishSingleRecord(record);
```

Comment:
If there were to be a lot of messages sent during channel suspension (some suggest as low as 7000), we could hit a max recursion depth.

Have we considered the risk of suspension during high-throughput, should we make it so that this method isn't reliant on recursion to avoid this issue?

Reply:
Agreed, recursion here is surely going to lead to trouble

Reply:
Thank you both. I agree that stack size could cause trouble here so I'll remove recursion from this call. But also I did think about putting boundary to the queue which will effect stack size here. However I didn't think that a typical use case would require adding significantly large amount of records between suspension and reconnection. Because if the connection fails, the connector will no longer be alive. 

I think if the above case is possible I can also add a boundary to the queue but I'm not really sure what the right size should be for that. After all there is a limit on how many messages can be published in a second so I'm thinking whether we should put a boundary based on that (for example 100). In this case we also need to decide what to do if this queue fills up. WDYT?

Reply:
Removed recursive call 6e7deac3d7ecc55bfa0867aa074e27c51e1c9933

Reply:
Just to check - with the recursive call removed, it looks like it now won't publish the original message?

Reply:
Yes indeed, the passing tests gave me a false confidence. I think I'll review tests later

Reply:
@AndyTWF Do you want to have another look at this since many change has been done today